client/transport/sse.go

```diff
@@ -112,13 +129,34 @@ func (c *SSE) Start(ctx context.Context) error {
 		}
 	}
 
+	// Add OAuth authorization if configured
+	if c.oauthHandler != nil {
+		authHeader, err := c.oauthHandler.GetAuthorizationHeader(ctx)
+		if err != nil {
+			// If we get an authorization error, return a specific error that can be handled by the client
+			if err.Error() == "no valid token available, authorization required" {
```

Comment:
I'm sorry, i just dig source code of this library and found this... WTF? 😱

I mean... Bro... Brooooo...